packages/server/utils/ServerHealthChecker.ts

```diff
@@ -64,8 +65,8 @@ export default class ServerHealthChecker {
           const connections = record[1]
           return connections.map((connection) => {
             const presence = JSON.parse(connection) as UserPresence
-            const {socketServerId, socketId} = presence
-            if (!deadServerIds.includes(socketServerId)) return
```

Comment:
-1 I think how we're detecting dead servers is wrong. We're only considering server present in socketServers
![Screenshot from 2023-06-26 12-08-43](https://github.com/ParabolInc/parabol/assets/7331043/8b8b3f51-d6f4-488b-bb79-aec7050e1937)
but we can still have stray presence entries
![Screenshot from 2023-06-26 12-08-54](https://github.com/ParabolInc/parabol/assets/7331043/1c6b9094-7206-4e4b-b480-296778610019)
which will consequently never get cleaned up.
I think the best course of action would be to use `socketServers` to ping, construct a list of alive servers and remove all presence entries which are not in the alive servers list. This will also take care of old entries which have `socketInstanceId === undefined`

Reply:
this sounds great! i much prefer `include` logic vs. `exclude`. will fix tomorrow